src/postgres/commands/configurePostgresFirewall.ts

```diff
@@ -32,8 +31,8 @@ export async function configurePostgresFirewall(context: IActionContext, treeIte
 export async function setFirewallRule(treeItem: PostgresServerTreeItem, ip: string): Promise<void> {
 
     const client: PostgreSQLManagementClient = createAzureClient(treeItem.root, PostgreSQLManagementClient);
-    const resourceGroup: string = azureUtils.getResourceGroupFromId(treeItem.id);
-    const serverName: string = nonNullProp(treeItem.server, 'name');
+    const resourceGroup: string = nonNullProp(treeItem, 'resourceGroup');
+    const serverName: string = nonNullProp(treeItem, 'label');
```

Comment:
You should only use `label` for stuff displayed to the user (like the message above is fine). For this, you should make `_azureName` public and use that. There's a few other instances like this throughout the PR

Reply:
azureName can only be used in cases of firewall or credentials related code. Label is necessary for other cases as the naming would differ between attached and azure accounts.

Reply:
I can add the name() function back to the ServerTreeItem instead of using azureName directly in few places and label in others.

Reply:
Changed label to azureName for non UI use cases.